detector/tracker/TrackerEndcap_o2_v06_geo.cpp

```diff
@@ -184,11 +184,11 @@ static Ref_t create_detector(Detector& theDetector, xml_h e, SensitiveDetector s
             
             Box mod_shape(m_vol.solid());
             
-             if(r - mod_shape->GetDZ()<innerR)
-                 innerR= r- mod_shape->GetDZ();
+             if(r < innerR)
+                 innerR = r;
```

Comment:
A bit pedantic, these cases can be stated like `innerR = std::min(innerR, r)` which I think better shows the intention.